File: guides/java/kora-java-guide-s3-app/src/main/java/ru/tinkoff/kora/guide/s3/repository/UserRepository.java
package ru.tinkoff.kora.guide.s3.repository;

import java.util.List;
import java.util.Optional;
import ru.tinkoff.kora.guide.s3.dto.UserResponse;

public interface UserRepository {

    List<UserResponse> findAll();

    Optional<UserResponse> findById(String id);

    String save(String name, String email);

    boolean update(String id, String name, String email);

    boolean deleteById(String id);
}

File: guides/kotlin/kora-kotlin-guide-grpc-server-app/src/main/kotlin/ru/tinkoff/kora/guide/grpcserver/repository/UserRepository.kt
package ru.tinkoff.kora.guide.grpcserver.repository

import ru.tinkoff.kora.guide.grpcserver.dto.UserResponse

interface UserRepository {
    fun findAll(): List<UserResponse>

    fun findById(id: String): UserResponse?

    fun save(name: String, email: String): String

    fun update(id: String, name: String, email: String): Boolean

    fun deleteById(id: String): Boolean
}
